### PR TITLE
clarifying security patches

### LINKED ISSUE #3521: Document that dbt Cloud-supported versions get security patches
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/docs/dbt-versions/core

### What part(s) of the page would you like to see updated?

Per [this internal conversation](https://dbt-labs.slack.com/archives/C0131TY7EEA/p1686668217878079), let's make sure to mention that Cloud supported versions will get security patches. 

Some questions for @jtcohen6:
* Should we refine [these definitions](https://docs.getdbt.com/docs/dbt-versions/core#ongoing-patches) to represent the event that any version is also a Cloud supported version?
* Should we add an asterisk to the Cloud-supported versions and a note mentioning they are also getting security fixes?
* Should we add an EOL definition?

For the writer addressing this issue:
* Reminder: the Releases table also shows up on the [upgrading core in cloud](https://docs.getdbt.com/docs/dbt-versions/upgrade-core-in-cloud#latest-releases) page, so we should make sure to add any of this context over there too (or add it to the reusable).

 

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mirnawong1**
clarify language about security patches in eol, added some definitions to support levels, and update v0.x to 'deprecated'. 

Resolves #3521

i think we may need to reorg this page and also replace eol headed with a specific backporting header to detail our backporting policy.

### CODE REVIEW

File: website/docs/docs/dbt-versions/core-versions.md
Comment (mirnawong1) [inline website/docs/docs/dbt-versions/core-versions.md]: i added this here bc i thought it would be a nice intro into the types of supported versions but commented out.

File: website/docs/docs/dbt-versions/core-versions.md
Comment (mirnawong1) [inline website/docs/docs/dbt-versions/core-versions.md:26]: longer term, i think we should replace this eol section with a ##backporting policy header and explain our policy there.

File: website/docs/docs/dbt-versions/core-versions.md
Comment (mirnawong1) [inline website/docs/docs/dbt-versions/core-versions.md]:
```suggestion
All dbt Core versions with an end-of-life (EOL) support level will no longer receive bug fixes.  We have been backporting certain changes to the 1.0.latest and 1.1.latest branches of dbt-core, so that we can continue running automated tests, while those versions continue to be available in dbt Cloud.

We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/core/installation#upgrading-dbt-core) to continue receiving bug fix support. 
```

File: website/docs/docs/dbt-versions/core-versions.md
Comment (mirnawong1) [inline website/docs/docs/dbt-versions/core-versions.md]:
```suggestion
| End of Life  | No longer receiving bug fixes and limited to receiving only security patches. <br /><br />All EOL dbt Core versions v1.0 and later are available in dbt Cloud until further notice. We plan to align dbt Cloud availability with dbt Core support and you'll receive advance notice before any changes take place. |
```

File: website/docs/docs/dbt-versions/core-versions.md
Comment (mirnawong1) [inline website/docs/docs/dbt-versions/core-versions.md]:
```suggestion
All dbt Core versions with an end-of-life (EOL) support level will no longer receive bug fixes.  
```

File: website/docs/docs/dbt-versions/core-versions.md
Comment (mirnawong1) [inline website/docs/docs/dbt-versions/core-versions.md]:
```suggestion
All dbt Core versions v1.0 and later are available in dbt Cloud until further notice. In the future, we intend to align dbt Cloud availability with dbt Core ongoing support. You will receive plenty of advance notice before any changes take place.

We have been backporting certain changes to the 1.0.latest and 1.1.latest branches of dbt-core, so that we can continue running automated tests, while those versions continue to be available in dbt Cloud.
```

File: website/docs/docs/dbt-versions/core-versions.md
Comment (jtcohen6) [reply website/docs/docs/dbt-versions/core-versions.md:26]: IMO "backporting" describes an implementation detail (our git branching strategy within the `dbt-core` repo), whereas "end of life" and "ongoing support" are the intended outcome we're looking to achieve & communicate here

There should be a correspondence between "what do we (Core team) choose to backport" and "what types of fixes are included in most recent minor version (active support), versus all pre-EOL versions" — but the target audience for each feels slightly different.

File: website/docs/docs/dbt-versions/core-versions.md
Comment (jtcohen6) [inline website/docs/docs/dbt-versions/core-versions.md]: A few notes:
- "Active" support has become a bit more active than in the past. We will patch regressions, new bugs, and even include fixes for some older bugs / quality-of-life improvements, as long as we have high confidence that those changes are narrowly scoped and will not cause unintended side effects.
- "End of Life": Minor versions that have reached EOL do not get **any** new patch releases. If it's an issue with dependencies, we resolve this within dbt Cloud. We do not resolve it by putting out a new patch release. So this sentence is not correct: "limited to receiving only security patches."
- "Deprecated": We're not talking about features here, we're talking about older releases of dbt Core. Given that the current rule applies to all pre-1.0 (v0.X) versions, should we just say that explicitly here?
> **Deprecated:** dbt-core versions older than v1.0 are no longer maintained by dbt Labs, nor supported in dbt Cloud.

File: website/docs/docs/dbt-versions/core-versions.md
Comment (jtcohen6) [inline website/docs/docs/dbt-versions/core-versions.md]: > All dbt Core versions with an end-of-life (EOL) support level will no longer receive bug fixes.

We can make this stronger: They'll no longer receive any fixes in new patch releases.

File: website/docs/docs/dbt-versions/core-versions.md
Comment (jtcohen6) [inline website/docs/docs/dbt-versions/core-versions.md]:
```suggestion
```
This is our team's choice for now. I don't think we should put it here, as part of the public policy.

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (jtcohen6) [inline website/docs/docs/dbt-versions/upgrade-core-in-cloud.md]: How about something like:
```suggestion
> **Starting with v1.0, all subsequent minor versions are available in dbt Cloud. Versions are actively supported, with patches and bug fixes, for 1 year after their initial release. At the end of the 1-year window, we encourage all users to upgrade to a newer version, for better ongoing maintenance and support.**
```

File: website/docs/docs/dbt-versions/core-versions.md
Comment (jtcohen6) [reply website/docs/docs/dbt-versions/core-versions.md]: Also: Should we snippetize this section, so it can be reused in the other page (without copy-paste)?

File: website/snippets/core-versions-table.md
Comment (jtcohen6) [inline website/snippets/core-versions-table.md]: How about something like:
```suggestion
_*All versions of dbt Core since v1.0 are available in dbt Cloud until further notice. Versions that are EOL do not receive any fixes. For the best support, we recommend upgrading to a version released within the past 12 months._
```

**Review by jtcohen6 — COMMENTED** [review]
Thanks for opening this @mirnawong1! I left some comments with recommended changes to the language. The most important thing we need to clarify is that these two things are both true simultaneously:
- dbt Core versions >12 months old do not get any new fixes (in the form of patch releases).
- Those versions are still available in dbt Cloud until further notice

Just because they're available, doesn't mean they're well-supported. We should strongly, strongly recommend that users upgrade for best results & best ongoing support — even if we're not yet in a place of regularly forcing migrations.

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (mirnawong1) [reply website/docs/docs/dbt-versions/upgrade-core-in-cloud.md]: fixed typo 'initial'

File: website/docs/docs/dbt-versions/core-versions.md
Comment (mirnawong1) [inline website/docs/docs/dbt-versions/core-versions.md]:
```suggestion
All dbt Core versions with an end-of-life (EOL) support level will no longer receive any fixes in new patch releases. We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/docs/core/installation#upgrading-dbt-core) to continue receiving support. 
```

File: website/docs/docs/dbt-versions/core-versions.md
Comment (mirnawong1) [reply website/docs/docs/dbt-versions/core-versions.md]: thanks @jtcohen6 ! ok so to clarify the issue and [slack thread](https://dbt-labs.slack.com/archives/C0131TY7EEA/p1686693075613039?thread_ts=1686668217.878079&cid=C0131TY7EEA), which seems to focus on security patches,  is the below correct? how do we 'address this within dbt Cloud'?:

> Minor versions that have reached EOL no longer receive new patch releases. If there's an issue with security or dependencies, we address this within dbt Cloud rather than through a new patch release. 


and yea ill try to create a snippet now. i thought it wasn't going to be much work since its only two places but best to change it in one place. also the links are formatted differently depending on what page this is on but i can add the full link for all

Comment (jtcohen6) [reply website/docs/docs/dbt-versions/core-versions.md]: IMO let's just cut the second sentence, and leave this as:

> Minor versions that have reached EOL no longer receive new patch releases.

We continue to make those versions available in dbt Cloud. Occasionally, that requires us to pin certain upstream dependencies while building images for older versions within the dbt Cloud application. That's our implementation detail; I don't think that's something we need to document as part of this policy.

(& totally your call on whether the snippet is worth it or not!)

File: website/snippets/core-version-support.md
Comment (mirnawong1) [inline website/snippets/core-version-support.md]:
```suggestion
- **[End of Life](/docs/dbt-versions/core#eol-version-support)** &mdash; Minor versions that have reached EOL no longer receive new patch releases.
```

**Comment by mirnawong1** [comment]
thanks @jtcohen6 , i've finalized that last change, can you rereview once more? [deploy preview here](https://deploy-preview-3532--docs-getdbt-com.netlify.app/docs/dbt-versions/core) (i think its still building though)

File: website/docs/docs/dbt-versions/core-versions.md
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/core-versions.md]:
```suggestion
dbt Labs provides different support levels for different versions, which may include new features, bug fixes, or security patches:
```

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/upgrade-core-in-cloud.md]:
```suggestion
> **Starting with v1.0, all subsequent minor versions are available in dbt Cloud. Versions are actively supported, with patches and bug fixes, for 1 year after their initial release. At the end of the 1-year window, we encourage all users to upgrade to a newer version for better ongoing maintenance and support.**
```

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/upgrade-core-in-cloud.md]:
```suggestion
We provide different support levels for different versions, which may include new features, bug fixes, or security patches:
```

File: website/docs/docs/dbt-versions/upgrade-core-in-cloud.md
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/upgrade-core-in-cloud.md]:
```suggestion
dbt Labs has always encouraged users to upgrade dbt Core versions whenever a new minor version is released. We released our first major version of dbt - `dbt 1.0` - in December 2021. Alongside this release, we updated our policy on which versions of dbt Core we will support in dbt Cloud.
```

File: website/docs/docs/dbt-versions/core-versions.md
Comment (matthewshaver) [inline website/docs/docs/dbt-versions/core-versions.md]: I think it's important to call out that it's bug fixes. We will provide security fixes and patches where required. We also occasionally backport non-bug fixes that address backwards compatibility. 
```suggestion
All dbt Core minor versions that have reached end-of-life (EOL) will have no new patch releases. This means they will no longer receive any fixes, including for known bugs that have been identified. Fixes for those bugs will instead be made in newer minor versions that are still under active support.

We recommend upgrading to a newer version in [dbt Cloud](/docs/dbt-versions/upgrade-core-in-cloud) or [dbt Core](/docs/core/installation#upgrading-dbt-core) to continue receiving support. 
```

Comment (mirnawong1) [reply website/docs/docs/dbt-versions/core-versions.md]: good question @matthewshaver, we originally had bug fixes but @jtcohen6 recommended stronger language [here](https://github.com/dbt-labs/docs.getdbt.com/pull/3532#discussion_r1234213710): "We can make this stronger: They'll no longer receive any fixes in new patch releases."

I do think including bug fixes is helpful. @jtcohen6 @matthewshaver, what about:

>All dbt Core versions with an end-of-life (EOL) support level will no longer receive any fixes, including bug fixes, in new patch releases.

Comment (jtcohen6) [reply website/docs/docs/dbt-versions/core-versions.md]: That statement is accurate. Here's how I think about it:

> All dbt Core minor versions that have reached end-of-life (EOL) will have no new patch releases. This means they will no longer receive any fixes, including for known bugs that have been identified. Fixes for those bugs will instead be made in newer minor versions that are still under active support.

A lot of passive voice, and a bit redundant — but I think it's important that we drive this point home.

Comment (mirnawong1) [reply website/docs/docs/dbt-versions/core-versions.md]: sounds great thank you @jtcohen6 and i dont think it's redundant at all! it's very important to very clear about this, i always remember clarifying things that might seem 'obvious' to seasoned people but not so obvious to others so i always like to side with being extra clear.

**Comment by mirnawong1** [comment]
hey @matthewshaver and @jtcohen6 , asking for a final rereview - let me know what you think and I've enabled auto merge if it's all looking ok!

File: website/snippets/core-version-support.md
Comment (jtcohen6) [inline website/snippets/core-version-support.md:3]:
```suggestion
- **[Critical](/docs/dbt-versions/core#ongoing-patches)** &mdash; When a new minor version is released, the previous minor version will move from "Active" to "Critical" support. These versions will still get smaller and less frequent patches, containing security and installation fixes.
```

**Review by jtcohen6 — APPROVED** [review]
LGTM! Thanks for this @mirnawong1!

One small suggestion for the wording of "critical" support, just to clarify intent; you can reword or reject.

**Pull request merged**
